app/src/test/java/io/openshift/booster/SsoIT.java

```diff
@@ -0,0 +1,227 @@
+                .map(r -> "https://" + r.getSpec().getHost() + "/auth")
+                .orElseThrow(() -> new IllegalStateException("Couldn't find secure-sso route"));
+
+        System.setProperty("sso.auth.server.url", ssoAuthUrl);
+
+        authzClient = createAuthzCLient(ssoAuthUrl);
+
+        applicationUrls = routes.stream()
+                .filter(r -> r.getMetadata().getName().contains("secured"))
+                .map(r -> "http://" + r.getSpec().getHost())
+                .collect(toList());
+
+    }
+
+    private static Greeting getGreeting(String url, String token, String from) {
+        Client client = ClientBuilder.newClient();
+        try {
+            WebTarget target = client.target(url);
+            System.out.println("URL : " + url);
+            target.register((ClientRequestFilter) requestContext -> {
+                requestContext.getHeaders().add("Authorization", "Bearer " + token);
+            });
+            IGreeting greetingClient = ((ResteasyWebTarget) target).proxy(IGreeting.class);
+            return greetingClient.greeting(from);
+        } finally {
+            client.close();
+        }
+    }
+
+    @Test
+    public void defaultUser_defaultFrom() {
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "password");
+        for (String url : applicationUrls) {
+            Greeting greeting = getGreeting(url, accessTokenResponse.getToken(), null);
+
+            assertThat(greeting).isNotNull();
+            assertThat(greeting.getContent()).contains("Hello, World!");
+        }
+    }
+
+    @Test
+    public void defaultUser_customFrom() {
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "password");
+        for (String url : applicationUrls) {
+            Greeting greeting = getGreeting(url, accessTokenResponse.getToken(), "Scott");
+
+            assertThat(greeting).isNotNull();
+            assertThat(greeting.getContent()).contains("Hello, Scott!");
+        }
+    }
+
+    // This test checks the "authenticated, but not authorized" flow.
+    @Test
+    @Ignore
+    public void adminUser() {
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("admin", "admin");
+        for (String url : applicationUrls) {
+            try {
+                getGreeting(url, accessTokenResponse.getToken(), null);
+                fail("ForbiddenException expected");
+            } catch (ForbiddenException e) {
+                // expected
+            }
+        }
+    }
+
+    @Test
+    public void badPassword() {
+        try {
+            AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "bad");
+            fail("HttpResponseException expected (401)");
+        } catch (Throwable t) {
+            assertThat(t).isInstanceOf(HttpResponseException.class);
+            assertThat(((HttpResponseException)t).getStatusCode()).isEqualTo(401);
```

Comment:
I think it should be `NotAuthorizedException` -- could check that similarly to `adminUser`. But this is good too.